applications/ShapeOptimizationApplication/python_scripts/data_logger_factory.py

```diff
@@ -73,6 +73,17 @@ def __CreateValueLogger( self ):
 
     # -----------------------------------------------------------------------------
     def __CreateDesignLogger( self ):
+        valid_output_modes = ["WriteDesignSurface", "WriteOptimizationModelPart", "None"]
```

Comment:
```suggestion
        valid_output_modes = ["WriteDesignSurface", "WriteOptimizationModelPart", "write_design_surface","write_optimization_model_part", "none"]
```

this way you can make it backward compatible